neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -1094,30 +1094,22 @@ object Neo4jStoreSingleton : GraphStore {
                     }
                 }
 
-                /* TODO: (kmm) The logic behind using region-code to fetch 'tax-rates' will fail
-                         when customers are linked to multiple regions. Currently it is assumed
-                         that a customer belongs to only one region. */
-                /* Use 'region-code' as the 'tax-region'. */
-                val region = customerStore.getRelated(customer.id, customerRegionRelation, transaction)
-                        .flatMap {
-                            if (!it.isEmpty())
-                                it.first().right()
-                            else
-                                NotFoundError(type = customerRegionRelation.name, id = "No region found for ${customer.id}")
-                                        .left()
-                        }
-                        .mapLeft {
-                            BadGatewayError("No region found for ${customer.id}", error = it)
-                        }
-                        .bind()
+                /* Fetch 'tax' id to be applied. */
+                val taxRegionId = if (product.payment.containsKey("taxRegionId"))
+                    product.payment["taxRegionId"]
+                else if (product.price.amount > 0)
+                    BadGatewayError("Product ${product.sku} has price ${product.price.amount} ${product.price.currency} but no 'taxRegionId' set")
```

Comment:
In next PR handle the case where the Product has price.amount > 0 but does not have a Tax associated with it.